### PR TITLE
Read navigation response on MSIDOAuth2EmbeddedWebviewController

### PR DESCRIPTION
**Pull request by juan-arias**
## Proposed changes

Allow reading navigation response from MSIDOAuth2EmbeddedWebviewController.
This is required to extract request-id & session-id from the response headers, this information will help troubleshoot when doing compliance check fails.

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/webview/embeddedWebview/MSIDOAuth2EmbeddedWebviewController.h [hunk 1]
```diff
@@ -34,6 +34,8 @@
 #import "MSIDAuthorizeWebRequestConfiguration.h"
 #import "MSIDWebViewPlatformParams.h"
 
+typedef void (^NavigationResponseBlock)(NSHTTPURLResponse *response);
```
Comment (oldalton) [inline IdentityCore/src/webview/embeddedWebview/MSIDOAuth2EmbeddedWebviewController.h:37]: nit: prefix with MSID, e.g. MSIDNavigationResponseBlock

**Review by oldalton — APPROVED** [review]

File: IdentityCore/src/webview/embeddedWebview/MSIDOAuth2EmbeddedWebviewController.m [hunk 2]
```diff
@@ -315,6 +315,20 @@ - (void)webView:(WKWebView *)webView didReceiveAuthenticationChallenge:(NSURLAut
...
 }
 
+- (void)webView:(WKWebView *)webView decidePolicyForNavigationResponse:(WKNavigationResponse *)navigationResponse decisionHandler:(void (^)(WKNavigationResponsePolicy))decisionHandler
+{
+    if (_navigationResponseBlock && navigationResponse && navigationResponse.response)
+    {
+        NSHTTPURLResponse *response = (NSHTTPURLResponse *)navigationResponse.response;
+        if (response)
+        {
+            self.navigationResponseBlock(response);
```
Comment (oldalton) [inline IdentityCore/src/webview/embeddedWebview/MSIDOAuth2EmbeddedWebviewController.m:325]: You're using "self.navigationResponseBlock" and "_navigationResponseBlock" above. Can you update to consistently use self.navigationResponseBlock?

File: IdentityCore/src/webview/embeddedWebview/MSIDOAuth2EmbeddedWebviewController.h (hunk 1 above)
Comment (oldalton) [reply IdentityCore/src/webview/embeddedWebview/MSIDOAuth2EmbeddedWebviewController.h]: Thanks, looking good.

**Review by oldalton — APPROVED** [review]
Added one more minor comment

File: IdentityCore/src/webview/embeddedWebview/MSIDOAuth2EmbeddedWebviewController.m (hunk 2 above)
Comment (juan-arias) [reply IdentityCore/src/webview/embeddedWebview/MSIDOAuth2EmbeddedWebviewController.m:325]: I got a warning when I first added it as `self.navigationResponseBlock`, but I no longer see it (and don't remember what was the previous warning). Updated.

**Review by oldalton — APPROVED** [review]

**Pull request merged**